src/pages/list/List.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from "react";
+import ListHeader from "./ListHeader";
+import RoomSortLine from "./RoomSortLine";
+import RoomLists from "./RoomLists";
+import ListFiltering from "./ListFiltering";
+import { getDistance } from "geolib";
+
+import * as S from "./List.styled";
+const List = () => {
+  const [roomList, setRoomList] = useState([]);
+  const [realList, setRealList] = useState();
+  const [isSelect, setIsSelect] = useState([true, false, false]);
+  const [filterings, setFilterings] = useState({
+    theme: "",
+    price: "100",
+    distance: "100000",
+  });
+  const [myPosition, setMyPosition] = useState({ lati: "", long: "" });
+
+  const getPosition = () => {
+    return new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition(resolve, reject);
+    });
+  };
+
+  async function getLoc() {
+    let obj = { ...myPosition };
+    try {
+      let position = await getPosition();
+
+      obj.lati = position.coords.latitude;
+      obj.long = position.coords.longitude;
+
+      setMyPosition(obj);
+    } catch (err) {
+      console.warn(`ERROR(${err.code}): ${err.message}`);
+
+      obj.userLat = 37.50591;
+      obj.userLng = 127.05401;
+
+      setMyPosition(obj);
+    }
+  }
+
+  useEffect(() => {
+    fetch("data/listData.json", {
+      method: "GET",
+    })
+      .then(res => res.json())
+      .then(data => {
+        getLoc();
+        data.forEach(list => {
+          list.distance = getDistance(
+            {
+              // latitude: myPosition.lati,
+              // longitude: myPosition.long,
+              latitude: 37.50591,
+              longitude: 127.05401,
+            },
+            {
+              latitude: Number(list.latitude),
+              longitude: Number(list.longtitude),
+            }
+          );
+        });
+
+        data.sort((a, b) => a.distance - b.distance);
```

Comment:
sort 옵션에 따라 리스트를 정렬해주는 로직을 따로 구현해둔다면
```suggestion
        setSelectedSort("거리순")
```